php/receipt.php

```diff
@@ -1,79 +1,52 @@
-        if (isset($_FILES['photo']['name']) && ($_FILES['photo']['name'] != '')) {
-          // On récupère le nom de l'image
-          $picture = $_FILES['photo']['name'];
-          // répertoire de stockage des images
-          $target = "pictures/".basename($picture);
-        } else  {
-          $picture = $_POST['uploadSrc'];
-        }
-        $date = $_POST['date'];
-        $receiptCategory =  $_POST['receiptCategory'];
-        $provider = mysqli_real_escape_string($conn, $_POST['provider']);
-        $amountTTC = $_POST['amountTTC'];
-        $tva = $_POST['tva'];
-        $_POST['ischecked'] === "true" ? $isChecked = 1 : $isChecked = 0;
-        if (isset($_POST['checkReceiptAndSelectNext'])) {
-          $isChecked = 1;
-        }
-        $description = mysqli_real_escape_string($conn, $_POST['description']);
-        if ($_POST['receiptid'] == "") { 
-          $query = "INSERT INTO `receipts`(`photo`, `date_emission`, `category`, `provider`, `montant_ttc`, `tva`, `checked`, `description`) VALUES ('$picture','$date','$receiptCategory', '$provider', $amountTTC,'$tva','$isChecked','$description')";
-          $result = mysqli_query($conn,$query);
-        }else { 
-          $id = $_POST['receiptid'];
-          $query = "UPDATE `receipts` SET `photo`='$picture', `date_emission`='$date', `category`='$receiptCategory', `provider`='$provider', `montant_ttc`=$amountTTC, `tva`='$tva', `checked`='$isChecked', `description`='$description' WHERE `receipts`.id = '$id'";
-          $result = mysqli_query($conn,$query);
-        }
-        
-        // On importe l'image
-        if (isset($_FILES['photo']['name']) && ($_FILES['photo']['name'] != '')) {
-          if (move_uploaded_file($_FILES['photo']['tmp_name'], $target)) {
-            sendMessage("L'import de l'image n'a pas fonctionné !", "danger");
-          }
-        }
 
-        if ($result) {
-          $msg = sendMessage("Ticket enregistré ! ", "success");
-        }else{
-          $msg = sendMessage("Ça n'a pas fonctionné ! ", "danger");
-        }
+    // ============================================================================
+    // saveReceipt ================================================================
+    // ============================================================================
+
+    if (isset($_POST['upload']) || isset($_POST['checkReceiptAndSelectNext'])) {
+        $result = $db->saveReceipt();
 
-      // if button check and next was clicked we go to the next receipt to check
-      if (isset($_POST['checkReceiptAndSelectNext'])) {
-        $nextReceiptToCheck = getFirstReceiptToCheck($conn);
+    // if saveReceipt worked  and button check and next was clicked we go to the next receipt to check
+    if ($result && isset($_POST['checkReceiptAndSelectNext'])) {
+        $nextReceiptToCheck = $db->getFirstReceiptToCheck();
+        $receiptCategories = ["restaurant", "gasoil", "hôtel", "péage", "autre"];  
```

Comment:
If your categories are the same everywhere, you should create a shared constant somewhere, and use it directly inside `getLinkWithParamsFromRow` so you avoid possible bugs and code duplcation.